.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,23 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/universal
+{
+	"name": "",
+
+	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+	"image": "ghcr.io/gpuweb/gpuweb:main",
+
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	// "features": {},
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "cp -r /grammar ./wgsl/"
```

Comment:
Since the installation of npm packages take considerable time and breaks from time to time, it is one of Docker image's primary strengths to cache that. I've added a comment line to note it.

Reply:
It's also in entrypoint.sh. Do we need both? Or is there another place we can put it that will subsume both?

Actually, do we even need to do this? If the image has npm's central cache in it, then installing again should be fast because npm can just copy the packages from its own cache.

Reply:
@kainino0x We need because entrypoint.sh is overridden by devcontainer by default, and I think it'd be just unnecessary overhead to start every dev environment with a full CI check (entrypoint.sh is the CI check of action).

For why, since the grammar script recognizes, this helps avoid creation of grammar.json etc., and makes sure network request is 100% not there.

Reply:
I see, makes sense. I was thinking more like, is there a place we can specify a post-create command in the Docker image instead of here, so it'll apply to both the devcontainer and the CI container, and run before the CI container's entrypoint? From a little reading it seems maybe the Dockerfile's ENTRYPOINT could be a script which (1) does this copy then (2) runs its arguments.

https://stackoverflow.com/a/34245657:
> The ENTRYPOINT specifies a command that will always be executed when the container starts.
> The CMD specifies arguments that will be fed to the ENTRYPOINT.


So we could do like this example:
https://code.visualstudio.com/remote/advancedcontainers/start-processes#_adding-startup-commands-to-the-docker-image-instead
except with the `echo "Hello from our entrypoint!"` replaced with the copy command. And perhaps with the default CMD set to perform the build instead of `sleep infinity`.

Reply:
@kainino0x is it possible to enable entrypoint in devcontainer.json so that this script only runs the very first time the dev environment is created rather than every launch? If that's possible, then cmd idea makes sense to me

Reply:
I don't know. Unless the copy command is really expensive, copying again each time the container is started seems fine. Or maybe we can symlink node_modules instead of copying?

Reply:
@kainino0x I have unified the execution by moving relevant parts to a prepare.sh script and running it in both entrypoint and devcontainer creation! Is this a potential way? It is in the latest commit.